src/Cake.Common/Tools/MSBuild/MSBuildRunner.cs

```diff
@@ -91,6 +98,17 @@ private ProcessArgumentBuilder GetArguments(FilePath solution, MSBuildSettings s
             return builder;
         }
 
+        private static string GetPlatformName(PlatformTarget platform)
+        {
+            switch (platform)
+            {
+                case PlatformTarget.MSIL:
+                    return "AnyCPU";                                    
+                default:
```

Comment:
I don't think that `ArgumentOutOfRangeException` fits well for this, since it's not an array type. Otherwise you could use `InvalidEnumArgumentException`.